tree/bst.go

```diff
@@ -160,3 +160,117 @@ func (n *node) getBSTNodeWithValue(val int) *node {
 
 	return n.right.getBSTNodeWithValue(val)
 }
+
+// changes By Prerna 08/01/2022
+
+// 1. Is validate BST
+func (t *BinarySearchTree) ValidBST() {
+	if t.isBST() {
+		fmt.Println("Valid BST")
+
+	} else {
+		fmt.Println("Invalid BST")
+	}
+}
+
+func (t *BinarySearchTree) isBST() bool {
+	if t == nil {
+		return true
+	}
+	return t.root.chkNode(100, 100)
+}
+func (n *node) chkNode(max int, min int) bool {
+	if n == nil {
+		return true
+	}
+	if n.val > max && n.val < min {
+		return false
+	}
+	if n.left.chkNode(n.val, min) && n.right.chkNode(max, n.val) {
+		return true
+	}
+	return false
+
+}
+
+// 2. determine if the given BST is balance
+
+func (t *BinarySearchTree) IsBal() {
+	if t == nil {
+		fmt.Println("Balanced")
+		return
+	}
+	h := 0
+	h1 := t.root.left.GetHght(h)
```

Comment:
Since, h is just used in the next two lines as initial value, directly pass 0 in both the places.